src/generator.cpp

```diff
@@ -1255,78 +1160,78 @@ std::string Generator::GeneratorImpl::generateOperatorCode(const std::string &op
     // 11. PIECEWISE (as an operator)                            [Right to left]
 
     if (isPlusOperator(ast)) {
-        if (isRelationalOperator(ast->leftChild())
-            || isLogicalOperator(ast->leftChild())
-            || isPiecewiseStatement(ast->leftChild())) {
-            leftChild = "(" + leftChild + ")";
+        if (isRelationalOperator(astLeftChild)
+            || isLogicalOperator(astLeftChild)
+            || isPiecewiseStatement(astLeftChild)) {
+            astLeftChildCode = "(" + astLeftChildCode + ")";
```

Comment:
Is it better to store the value of this method call?
```
bool relationalOperator = isRelationalOperator(astLeftChild);
```
rather than repeatedly making the same call over and over.

Reply:
and other calls to methods.

Reply:
We don't make the same call over and over, only once, depending on which operator `ast` is. So, if we were to do what you are suggesting then we would end up with calls that are not needed (e.g. the call to `isRelationalOperator(astLeftChild)` is not needed if `ast` is a root operator).